src/helpers/augmentedReality/deleteObject.js

```diff
@@ -0,0 +1,27 @@
+import AsyncStorage from '@react-native-async-storage/async-storage';
+import * as FileSystem from 'expo-file-system';
+
+import { DOWNLOAD_TYPE } from './downloadType';
+
+export const deleteObject = async ({ item, index, downloadableData }) => {
+  const { localUris, id: objectId } = item;
+  let newDownloadedData = [...downloadableData];
+
+  for (let i = 0; i < localUris.length; i++) {
+    const { downloadUri, title, type, id } = localUris[i];
+
+    try {
+      await FileSystem.deleteAsync(downloadUri);
+      await AsyncStorage.removeItem(`${objectId}-${title}${id}.${type}`);
+    } catch (e) {
+      console.error(e);
+    }
+  }
+
+  newDownloadedData[index].size = 0;
+  newDownloadedData[index].progressSize = 0;
+  newDownloadedData[index].DOWNLOAD_TYPE = DOWNLOAD_TYPE.DOWNLOADABLE;
+  newDownloadedData[index].localUris = [];
```

Comment:
I guess I don't know what you mean here. 🤷‍♂️

Reply:
OK, now I know what you mean. I'll work on it. 👍

Reply:
In order to inform the users, I have returned an alert in case of an error. In this way, we can show the users that the files are not deleted without updating the array shown on the screen.

https://github.com/ikuseiGmbH/smart-village-app-app/pull/554/commits/fa62822c3ee929b66da14c8b238c3648a0891aa6

Reply:
let's wait what happens after merging the two branches and resolving the conflicts.

Reply:
we solved the part for `deleteObject` and adjusted it with `checkDownloadedData` for consistency